src/jwt/mod.rs

```diff
@@ -20,11 +20,98 @@ new_type![
     JsonWebTokenContentType(String)
 ];
 
+#[derive(Error, Debug)]
+pub enum JsonWebTokenTypeNormalizeError {
+    #[error("Has parameters set without application specifies like JWT;param=true (should be application/JWT;param=true)")]
+    ParameterWithoutApplication(),
+
+    #[error(
+        "The web token type is complex (contains /) but does not start with an application type."
+    )]
+    ComplexTypeWithoutApplication(),
+}
+
 new_type![
+    /// JSON web token type field (typ)
+    ///
+    /// This types stores the raw (deserialized) value.
+    ///
+    /// To compare two different JSON Web Token types, please use the normalized version via [`JsonWebTokenType::normalize`]
     #[derive(Deserialize, Hash, Ord, PartialOrd, Serialize)]
```

Comment:
I'm noticing it's odd that `PartialOrd` and `Ord` are implemented but `PartialEq` and `Eq` aren't. If we don't want users to compare these non-normalized types to one another, then we should probably remove `PartialOrd`, `Ord`, and `Hash` too (not a breaking change since this type wasn't public before)

Reply:
I think they are implemented through https://github.com/ramosbugs/openidconnect-rs/blob/02525323bfdcef1c0da17835fa0d9738db354765/src/macros.rs#L161

But I removed `PartialOrd` and `Ord`.

Generally I also think that comparing these values shouldn't be (easily) possible as this might lead to wrong results. The more interesting derive would therefore be `Eq`, but as far as I can see the macro does not support switching off that specifically.